### DISCUSSION TITLE
#29

### DISCUSSION THREAD

**Discussion by verberg** [cssbattle/forum#192] [Q&A]
I have 3 solutions and can't get 100% match. Why?
Here my last one:
style>*{background:#F3AC3C}*>*{width:200;height:200;margin:50 auto;background:radial-gradient(#F3AC3C 106q,#1A4341 0 1q)25vw 25vw

**Comment by Beowolve** [comment]
In order to find the right numbers on this target, you need many hundred submissions, or you get very lucky. I have 450+ submissions so far and still no luck on the shorter versions.
`margin:50` auto can be replaced with `margin 50 100`, and you don't need width/height. Also, 1q on the end of radial-gradient can be deleted.

**Comment by SelenIT** [comment]
It's one of the most tricky targets that can require literally hundreds of attempts to get 💯. Apparently the scoring algorithm "doesn't like" sharp edges of gradients (rendered jagged on non-Retina screens) for places where curves are meeting at a sharp angle. Try smoothing the edges a bit, probably with fractional units (also try different units including `%` for it).

**Reply by gunnarbittersmann** [reply]
My first attempt was a 7.5% match with 125 characters. I had used `F3AC4` for `F3AC3C` – that was too far off.
2nd attempt with 126 characters: 💯.
That would currently be 41st rank. Shall I reveal that solution?

**Reply by SelenIT** [reply]
Was it a gradient-based solution?

**Reply by gunnarbittersmann** [reply]
No.

**Reply by SelenIT** [reply]
Well, probably I had to clarify that my comment was related to gradient-based solutions like the @verberg's. Other approaches may be easier to get 💯 (or not:).

**Reply by verberg** [reply]
Yes please, reveal your solution!
It's (for me) annoying to have different solutions and never knowing why it doesn't get 100%!

**Reply by gunnarbittersmann** [reply]
The solution I have revealed at [CSS Café](https://www.meetup.com/de-DE/CSS-Cafe/events/282467056/) the other day was (probably):
```
<body bgcolor=1A434 text=F3AC3C><img><style>img{border:2in solid;border-radius:25vw;margin:-242-192;box-shadow:4in 0,0 4in,4in 4in
```
 (130 characters)

Did I stop there because I was running out of time? This can be easily shortened by 2 characters, and then by another 2.

The video is not online yet – *mea culpa*. I’ll keep you updated (in [this thread](https://github.com/cssbattle/forum/discussions/197)).

**Reply by verberg** [reply]
wow, great!

**Reply by gunnarbittersmann** [reply]
During the presentation, I was going to show how this can be shortened further. Maybe I did and just didn’t hit the submit button?

Anyway, here we go: There’s just one rule in the stylesheet, so we might apply this to the `img` element inline:
```
<body bgcolor=1A434 text=F3AC3C><img style="border:2in solid;border-radius:25vw;margin:-242-192;box-shadow:4in 0,0 4in,4in 4in">
```

Quotation marks around attribute values are not needed in HTML when the value doesn’t contain certain characters, eg. spaces. Hm, we have spaces in there… But `0` and `4in` are the same as `+0` and `+4in` respectively. And it works without space before the `+`. (This might not be according to the spec, but who cares? 😏)

In the `border` declaration, we need to put the number after the border-style keyword:
```
<body bgcolor=1A434 text=F3AC3C><img style=border:solid+2in;border-radius:25vw;margin:-242-192;box-shadow:4in+0,0+4in,4in+4in>
```

**Reply by SelenIT** [reply]
Regarding gradient-based solution, basing on the code from the initial post, the following color stops seem to work: `#F3AC3C 105.3q,#1A4341 106q`. There can be shorter values in `%`s, but it might require a lot of patience to find them 😳
